### PR TITLE
Gamepad fixes; resolves #1699

### LINKED ISSUE #1699: Game pad + Oculus + XBox controller gives TypeErrors
This is with Oculus Rift + XBox controller + Game Pad patch:

1) Open ChromiumVR
2) Open default scene in edit mode
3) Add game pad patch
4) add a "Float debug" and connect it to "Left trigger".

Observe how the clock runs very fast, and if you open the console you see lots of TypeErrors.

### PR DESCRIPTION
**Pull request by kschzt**

### CODE REVIEW

File: browser/plugins/plugins.json [hunk 1]
```diff
@@ -208,7 +208,8 @@
         "Mouse buttons": "mouse_button_generator",
         "Mouse position": "mouse_position_generator",
         "Mouse wheel": "mouse_wheel_generator",
-        "Game pad": "gamepad_generator"
+        "XBOX 360 Gamepad": "gamepad_generator",
+        "HTC Vive Controller": "vive_controller"
```
Comment (gmarinov) [inline browser/plugins/plugins.json]: please add this in node-ui-enum.js (L162-171) too

**Comment by kkananen** [comment]
lgtm

File: documentation/browser/plugins/vive_controller.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+##Inputs
+###pad number
+Gamepad number
+
+###vibrate
+Vibrates the gamepad while this is true
+
+##Outputs
+###touchpad clicked
```
Comment (gmarinov) [inline documentation/browser/plugins/vive_controller.md]: 'down'

**Comment by kkananen** [comment]
lgtm

**Pull request merged**